### PR TITLE
fix(lib-injection): do not import user installed ddtrace

### PR DESCRIPTION
**Pull request by brettlangdon**
One of the checks we do when auto-injecting into a process is to see if the user already has `ddtrace` installed. If we do, then we will _not_ inject into it.

However, the way we check if the user has `ddtrace` installed is we `import ddtrace`. This may have an unintended consequence of importing `ddtrace` into a process which otherwise wouldn't normally.

The impact of importing `ddtrace` should be fairly low, since we won't instrument anything, but it is better to avoid the import overhead if the user wasn't going to import it anyways.

Instead we move to using `importlib.util.find_spec` to search for the module on the `sys.meta_path`. The downside to doing this is we won't know the installed `ddtrace` version.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: lib-injection/sources/sitecustomize.py [hunk 1]
```diff
@@ -205,9 +206,12 @@ def _inject():
     integration_incomp = False
     runtime_incomp = False
     os.environ["_DD_INJECT_WAS_ATTEMPTED"] = "true"
+    spec = None
     try:
-        import ddtrace
-    except ImportError:
+        spec = importlib.util.find_spec("ddtrace")
+        if not spec:
```
Comment (brettlangdon) [inline lib-injection/sources/sitecustomize.py:212]: getting `None` back is a valid response.

I decided to just raise in that case to keep majority of this code the same as it was previously.

**Comment by P403n1x87** [comment]
> The downside to doing this is we won't know the installed ddtrace version.

But once we have the spec we could potentially get hold of the `_version.py` source and retrieve the version from there, no?

**Comment by brettlangdon** [comment]
> Wouldn't we need unit or regression tests to validate this change?

Yes, but I have no idea where this regression test should go. I didn't see any lib-injection tests in this repo, so probably needs to be a system test change?

I can take a quick look into that today, I would like a regression test.

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**